src/io/flutter/run/SdkFields.java

```diff
@@ -129,6 +130,9 @@ public GeneralCommandLine createFlutterSdkRunCommand(@NotNull Project project,
     if (buildFlavor != null) {
       args = ArrayUtil.append(args, "--flavor=" + buildFlavor);
     }
+    if (FlutterSettings.getInstance().isShowStructuredErrors()) {
+      args = ArrayUtil.append(args, "--dart-define=flutter.inspector.structuredErrors=true");
```

Comment:
@DanTup for FYI

Reply:
Great, this should fix https://github.com/Dart-Code/Dart-Code/issues/1850.

I presume it's safe to pass `--dart-define=` to older SDKs and this won't need a version check?

Reply:
I'm not sure about this - is the concern that an older SDK will break when using `--dart-define` or this particular flag? If the SDK doesn't consider the flag yet there should be no problem, but I don't know if anything happens if `--dart-define` doesn't exist.

Reply:
> is the concern that an older SDK will break when using `--dart-define` or this particular flag?

Either, I guess. I wondered if it's safe for me to just always send this (if structured errors is enabled), or if that might result in errors for users with older Flutter SDKs. As long as it works in the latest stable and later it's probably not a real concern, but since I already have version checks on some flags, it's easy to add one if required to have compatibility with more SDKs.

(I can test it if nobody knows the answer off-hand, I just thought I'd ask and save the multiple sets of downloads changing between versions 😁)

Reply:
Yes can you test and update me too? If there is an issue I think I would need a version check as well.

Reply:
Looks like `--dart-define` exists in both the current stable, and the previous stable (v1.12), added in https://github.com/flutter/flutter/pull/44083.

I think I'll add a version in VS Code check since it's easy and I already have it for some other things, but it's probably not entirely necessary.